xmrstak/backend/cpu/minethd.cpp

```diff
@@ -618,9 +620,11 @@ void minethd::multiway_work_main(size_t N, cn_hash_fun_multi hash_fun_multi)
 				iTimestamp.store(iStamp, std::memory_order_relaxed);
 			}
 
-			if((nonce_ctr++ & (nonce_chunk/2 - 1)) == 0)
+			nonce_ctr -= N;
```

Comment:
Change looks good to me. Sorry I wasn't too familiar with the nonce calculation and was just guessing what to do. Thanks a lot!